src/borda_clj/client.clj

```diff
@@ -0,0 +1,90 @@
+(ns borda-clj.client
+  (:use [slingshot.slingshot :only [throw+ try+]])
+  (:require [aleph.http :as http]
+            [byte-streams :as bs]
+            [cheshire.core :as json]))
+
+(def submit-key {:op "_submit"})
+
+(defn submit-failed [measurements values]
+  (update measurements submit-key (partial merge-with +) {:error_count (get values :_submits)}))
+
+(defn collect
+  "Adds the given measurement (values and dimensions) to the given hashmap and
+   caps the size of the hashmap at the given max-buffer-size."
+  [measurements max-buffer-size dimensions values]
+  (let [key  (into (sorted-map) dimensions)
```

Comment:
Just curious: what's the rationale behind converting `dimensions` into a sorted-map to use it as a key? My guess was performance, but have you seen that recommended somewhere?

Reply:
Oh, great question. IIRC from Java land, if you used a hashmap as a key to a map, two hashmaps with the same contents wouldn't always be seen as equal. It occurs to me that Clojure is much smarter about that stuff, so I don't actually need it. Sure enough, tests pass without using a sorted map here.